source/visionEnhancementProviders/NVDAHighlighter.py

```diff
@@ -432,11 +439,10 @@ def _run(self):
 		try:
 			if vision._isDebug():
 				log.debug("Starting NVDAHighlighter thread")
-			try:
-				window = self._window = self.customWindowClass(self)

```

Comment:
Explanation of changes:

When an error was raised in customWindowClass constructor it was not caught here, though `_highlighterRunningEvent.set()` was called in the finally block. This then creates a race condition at line 423:
`if waitResult is False or not self._highlighterThread.is_alive():`. The thread may not have time to die before `is_alive` is checked. Then there is no exception raised.